test/test_image.py

```diff
@@ -0,0 +1,473 @@
+import os
+from shlex import split
+from subprocess import PIPE, CalledProcessError, run
+
+import pytest
+
+from docker_cli.exceptions import CommandNotFoundOnImageError
+from docker_cli.image import Image, get_image_id
+
+
+@pytest.fixture
+def test_image_id():
+    """
+    Builds an image for testing and returns its ID.
+    """
+    run(split("docker build ./ -t test"))
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    return inspect_process.stdout.strip()
```

Comment:
It's possible for pytest fixtures to perform some teardown operations in order to clean up after themselves when the test has finished running. We could use this to automatically clean up our test image when we're done with it.

The way to do this is by `yield`ing instead of `return`ing the fixture value, and then inserting the teardown code after the `yield statement.

https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended

```suggestion
    yield inspect_process.stdout.strip()
    run(split(f"docker image rm {tag}"))
```